src/module/Entity/config/types.config.php

```diff
@@ -164,23 +182,158 @@
                             ]
                         ]
                     ],
-                    'license'    => []
+                    'license' => []
                 ]
             ],
-            'text-solution'         => [
-                'title'       => 'id',
+            'input-string-normalized-match-challenge' => [
+                'title' => 'id',
                 'description' => 'content',
-                'components'  => [
+                'components' => [
                     'repository' => [
-                        'form'   => __NAMESPACE__ . '\Form\TextSolutionForm',
+                        'form' => __NAMESPACE__ . '\Form\InputChallengeForm',
+                        'fields' => [
+                            'solution',
+                            'feedback',
+                            'changes',
+                        ]
+                    ],
+                    'link' => [
+                        'children' => [
+                            'input-string-normalized-match-challenge' => [
+                                'multiple' => true
+                            ],
+                            'input-number-exact-match-challenge' => [
+                                'multiple' => true
+                            ],
+                            'input-expression-equal-match-challenge' => [
+                                'multiple' => true
+                            ]
+                        ],
+                        'parents' => [
+                            'text-exercise' => [
+                                'multiple' => false
+                            ],
+                            'grouped-text-exercise' => [
+                                'multiple' => false
+                            ],
+                            'input-string-normalized-match-challenge' => [
+                                'multiple' => false
+                            ],
+                            'input-number-exact-match-challenge' => [
+                                'multiple' => false
+                            ],
+                            'input-expression-equal-match-challenge' => [
+                                'multiple' => false
+                            ]
+                        ]
+                    ],
+                    'license' => []
+                ]
+            ],
+            'input-number-exact-match-challenge' => [
+                'title' => 'id',
+                'description' => 'content',
+                'components' => [
+                    'repository' => [
+                        'form' => __NAMESPACE__ . '\Form\InputChallengeForm',
```

Comment:
We use the same form for every `input-*-match-challenge` type. We only need custom forms if the `input-*-match-challenge` type accepts some kind of parameters.